analyses/upstream-analysis/README.md

```diff
@@ -0,0 +1,116 @@
+
+To run all of the Rscripts in this module sequentially on an interactive session on HPC, please run the following command from an interactive compute node:
+
+```
+bash run-upstream-analysis.sh
+```
+
+### Run module by using lsf on HPC
+
+There is also the option to run a lsf job on the HPC cluster by using the following command on an HPC node:
+
+```
+bsub < lsf-script.txt
+```
+
+
+## Folder content
+
+This folder contains scripts tasked to:
+(1) Infer QC metrics and associated plots to visually explore the quality of each library of the project.
+(2) Evaluate QC metrics and set filters to remove low quality cells in 10x single-cell- and single-nuclei-RNA-sequencing libraries (without cell hashing experiment).
+
+## QC Steps and methods
+
+The pipeline allows for the user to include/exclude methods and adjust the pipeline during QC according to the sequence type, expected cells number, type of experiment, and genome reference.
+
+If the user does not wish to include results from step (2), this can be defined in the `project_parameters.Config.yaml` file. 
+
+
+### (1) Seurat QC metrics
+
+[Seurat](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html) and [scooter](https://github.com/igordot/scooter) workflows are implemented to pre-process, filter and plot the RNA-sequencing data. The CellRanger output from the `cellranger-analysis` module is used for this step. User will have to define `params` as needed for their experiment. 
+  - Before and after filter: Plot distribution of the number of genes, UMI, and percent mitochondrial reads per cell.
+  - Summary of Cell Statistics: Percent of reads in cells, Median UMI count per cell, Median genes detected per cell, Median percent reads mitochondrial.
+  - Data were normalized by using the global-scaling normalization method “LogNormalize” that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result. Then, highly variable genes (HVGs) are selected to subset features that indicate high cell-to-cell variation in the dataset (i.e, they are highly expressed in some cells, and lowly expressed in others). Then, these HVGs are used as input to principal component analysis, and the top 30 principal components are selected. A combination of different dimensions (20, 25) and number of neighbors (30, 20, 10) are used along with the principal components to calculate the UMAP (Uniform Manifold Approximation and Projection) embeddings.
+  
+Here, the user can select to implement the following strategies to remove low quality cells:
+- `step A` [miQC](https://bioconductor.org/packages/devel/bioc/vignettes/miQC/inst/doc/miQC.html) R package. The miQC model is based on the assumption that there are a non-trivial number of compromised cells in the dataset, which is not true in all datasets. If it is already known that, we recommend that the user skip this step. 
```

Comment:
I feel that the last sentence here is a bit unclear. Maybe to be more clear it could say something to the effect of, "If it is already know that the dataset is high-quality with a trivial number of compromised cells, we recommend..."

Reply:
```suggestion
- `step A` [miQC](https://bioconductor.org/packages/devel/bioc/vignettes/miQC/inst/doc/miQC.html) R package. The miQC model is based on the assumption that there are a non-trivial number of compromised cells in the dataset, which is not true in all datasets. If it is already known that the dataset is high-quality with a trivial number of compromised cells, we recommend that the user skip this step. 
```